src/java_bytecode/java_types.h

```diff
@@ -507,6 +507,20 @@ class java_generic_symbol_typet : public symbol_typet
   {
     return (generic_typest &)(add(ID_generic_types).get_sub());
   }
+
+  /// Check if this symbol has the given generic type. If yes, return its index
+  /// in the vector of generic types.
+  /// \param type The type we are looking for.
+  /// \return The index of the type in the vector of generic types.
+  optionalt<size_t> generic_type(const reference_typet &type) const
```

Comment:
Is there a reason to return an index (not yet looked at TG PR)? It seems like this code at least would be simpler to just return a `optionalt<reference_typet>`

Reply:
I will need the index later on in test generator so I thought a separate function for it would be useful. Would changing the name of the function make it more clear?